src/ffi/index.ts

```diff
@@ -1,19 +1,159 @@
-import bindings = require('bindings');
+import path from 'node:path';
+import bindings = require('node-gyp-build');
 import logger, { DEFAULT_LOG_LEVEL } from '../logger';
 import { LogLevel } from '../logger/types';
 import { Ffi } from './types';
 
-const ffiLib: Ffi = bindings('pact.node');
-
 export const PACT_FFI_VERSION = '0.4.0';
 
+// supported prebuilds
+// darwin-arm64
+// darwin-x64
+// linux-arm64
+// linux-x64
+// win32-x64
+
+const supportedPlatforms = [
+  'darwin-arm64',
+  'darwin-x64',
+  'linux-arm64',
+  'linux-x64',
+  'win32-x64',
+];
+const platform = `${process.platform}-${process.arch}`;
+
+const supportedPlatformsMessage = [
+  'Supported platforms are: ',
+  ` - ${supportedPlatforms.join('\n - ')}`,
+].join('\n');
+const detectedMessage = `We detected your platform as: \n\n - ${platform}\n`;
+logger.info(detectedMessage);
+if (
+  !supportedPlatforms.includes(platform) &&
+  process.env['UNSUPPORTED_PLATFORM'] !== platform
+) {
+  logger.warn(supportedPlatformsMessage);
+  logger.warn(detectedMessage);
+  logger.error(`Unsupported platform: ${platform}`);
+  throw new Error(
+    `Unsupported platform: ${platform}, set $UNSUPPORTED_PLATFORM to ${platform} to override this check`
+  );
+}
+
+if (process.env['UNSUPPORTED_PLATFORM'] !== platform) {
+  logger.warn(
+    `You have set the environment variable UNSUPPORTED_PLATFORM to ${process.env['UNSUPPORTED_PLATFORM']}, this will override the detected platform of ${platform}`
+  );
+}
+
+const loadPathMessage = (bindingsPath: string) =>
+  `: loading native module from: \n\n - ${path.join(
+    bindingsPath,
+    'prebuilds',
+    platform
+  )} ${
+    process.env['PACT_NAPI_NODE_LOCATION']
+      ? `\n - source: PACT_NAPI_NODE_LOCATION \n - You must have a supported prebuild for your platform at this location in the path ${path.join(
+          process.env['PACT_NAPI_NODE_LOCATION'],
+          'prebuilds',
+          platform
+        )}`
+      : `\n   source: ${path.join(
+          bindingsPath,
+          'prebuilds',
+          platform
+        )} \n\n - You can override via PACT_NAPI_NODE_LOCATION\n`
+  }`;
+
+const bindingsResolver = (bindingsPath: string | undefined) =>
+  bindings(bindingsPath);
+
+const bindingPaths = [
+  path.resolve(
+    process.env['PACT_NAPI_NODE_LOCATION']?.toString() ?? path.resolve(),
+    path.resolve(__dirname, '..', '..')
+  ),
+];
+let ffiLib: Ffi;
+let loadPath: string | undefined;
+try {
+  bindingPaths.forEach((bindingPath) => {
+    try {
+      loadPath = bindingPath;
+      logger.info(
+        `Attempting to find pact native module ${loadPathMessage(bindingPath)}`
+      );
+      ffiLib = bindingsResolver(bindingPath);
+      if (ffiLib) {
```

Comment:
Ah! I think this is why it's failing - should this be:

```suggestion
      if (!ffiLib) {
```

At the moment, it seems to be finding the package and then throwing an error!

Reply:
whooops 🤦🏾